qt/applications/workbench/workbench/plotting/figurewindow.py

```diff
@@ -38,6 +40,28 @@ def __init__(self, canvas, parent=None):
         self.setAttribute(Qt.WA_DeleteOnClose, True)
         self.setWindowIcon(QIcon(':/images/MantidIcon.ico'))
 
+        # On Windows, setting the Workbench's main window as this window's
```

Comment:
I had one final thought. Could we put this code in some type of mixin so that other windows could use it by just inheriting, a bit like `ObservingView`?